### ISSUE TITLE
Add API to register frontend validation handler for Custom Fields API

### ISSUE THREAD

**Issue by senadir** [woothemes/woocommerce#42129]
Part of the Custom Fields API project, we want to evaluate and consider adding a way to register custom JS logic for validating fields and returning errors client side, we will test the limit of simple regex validation and attributes first before we dive here.

The main limitation here is JS transportability. PHP validation will run no matter where the Checkout is being hosted, but JS validation is only limited to the core Checkout block. Other instances where the field might be rendered and interacted with will not get this validation (External Checkout like WooPay, External Order creation screens like the Woo App, a POS, or editing the field somewhere else).

For this reason, frontend validation should be considered part of progressive enhancement and not part of the core experience of a field, and should be emittable (relaying on server validation or attributes).

**Comment by pierorocca** [comment]
This needs engineering input before we can confirm the final disposition.

@senadir, is the client side validation still a worth addition and should it wait for iAPI?
